addon/-private/system/model/model.js

```diff
@@ -120,8 +120,8 @@ var Model = Ember.Object.extend(Ember.Evented, {
     @type {Boolean}
     @readOnly
   */
-  hasDirtyAttributes: Ember.computed('currentState.isDirty', function() {
-    return this.get('currentState.isDirty');
+  hasDirtyAttributes: Ember.computed('currentState', function() {
+    return this.get('_internalModel.currentState.isDirty');
```

Comment:
Ah, now I see. Sorry for being slow here. Re-reading your comment makes it clear what you meant. Yes, please change it back to `this.get('_internalModel.currentState.isDirty')`. Thanks!